src/utils/coconut.js

```diff
@@ -0,0 +1,79 @@
+import 'brace/mode/python';
+
+export class CoconutHighlightRules extends window.ace.acequire(
+  'ace/mode/python_highlight_rules'
+).PythonHighlightRules {
+  constructor() {
+    super();
+
+    const keywords =
+      'and|as|assert|break|class|continue|def|del|elif|else|except|exec|' +
+      'finally|for|from|global|if|import|in|is|lambda|not|or|pass|print|' +
+      'raise|return|try|while|with|yield|async|await|data|' +
+      // Coconut-specific:
+      'match|case';
+    // TODO: Add rest
+
+    const builtinConstants =
+      'True|False|None|NotImplemented|Ellipsis|__debug__|';
+    // Coconut-specific:
+    // TODO: Add Coconut-specific builtin constants (if applicable?)
+
+    const builtinFunctions =
+      'abs|divmod|input|open|staticmethod|all|enumerate|int|ord|str|any|' +
+      'eval|isinstance|pow|sum|basestring|execfile|issubclass|print|super|' +
+      'binfile|iter|property|tuple|bool|filter|len|range|type|bytearray|' +
+      'float|list|raw_input|unichr|callable|format|locals|reduce|unicode|' +
+      'chr|frozenset|long|reload|vars|classmethod|getattr|map|repr|xrange|' +
+      'cmp|globals|max|reversed|zip|compile|hasattr|memoryview|round|' +
+      '__import__|complex|hash|min|set|apply|delattr|help|next|setattr|' +
+      'buffer|dict|hex|object|slice|coerce|dir|id|oct|sorted|intern|self|' +
+      // Coconut-specific
+      'reduce|takewhile|dropwhile|tee|count|makedata|consume|parallel_map|' +
+      'addpattern|recursive_iterator|concurrent_map|fmap|__fmap__|starmap|reiterable|' +
+      'scan|groupsof|py_chr|py_filter|py_hex|py_input|py_int|py_object|py_oct|' +
+      'py_open|py_print|py_range|py_xrange|py_str|py_map|py_zip|';
+
+    const keywordMapper = this.createKeywordMapper(
+      {
+        'invalid.deprecated': 'debugger',
+        'support.function': builtinFunctions,
+        'constant.language': builtinConstants,
+        keyword: keywords,
+      },
+      'identifier'
+    );
+
+    const pythonHighlightRules = this.getRules();
+    pythonHighlightRules.start = pythonHighlightRules.start.filter(
+      rule =>
+        rule.regex !== '[a-zA-Z_$][a-zA-Z0-9_$]*\\b' &&
+        rule.token !== 'keyword.operator'
+    );
+
+    this.$rules = {
+      ...pythonHighlightRules,
+      start: [
+        ...pythonHighlightRules.start,
+        {
+          token: keywordMapper,
+          regex: '[a-zA-Z_$][a-zA-Z0-9_$]*\\b',
+        },
+        {
+          token: 'keyword.operator',
+          regex:
+            '\\+|\\-|\\*|\\*\\*|\\/|\\/\\/|%|<<|>>|&|\\||\\^|~|<|>|<=|=>|==|!=|<>|=|' +
+            '@|',
```

Comment:
Remove the last `|`. It's causing integers to not get highlighted
<img width="76" alt="screen shot 2018-03-09 at 5 47 17 pm" src="https://user-images.githubusercontent.com/8571671/37236942-02fbaaaa-23c2-11e8-84da-8cd7e51212cb.png">

Plus, I'd suggest doing the same thing as `keywords` and define `operators` above and make a clear separation with a comment saying coconut specific.

Reply:
Okay for sure! I'll make the changes shortly. Just to clarify, are you saying that the new_operators, split them up into built-ins and coconut specific?

Reply:
Also, you should add comments specifying the separation between python-syntax regex and coconut-syntax regex.

Reply:
<img width="682" alt="screen shot 2018-03-09 at 4 25 44 pm" src="https://user-images.githubusercontent.com/35832643/37238324-993c5e54-23d8-11e8-87e4-1881f4a33d36.png">
How do I get rid of this warning? I am trying to include the coconut specific keywords such as r"\$" and I'm getting the warning above.

Reply:
Also, I'm assuming that this is the python built in regex which are just separated by the |. If this is the case, it's not highlighting any of the backslash symbols. 
![screen shot 2018-03-09 at 8 38 22 pm](https://user-images.githubusercontent.com/35832643/37238375-d97c84e8-23d9-11e8-8f0d-3710682395cf.png)

This seems like it's going to be a problem for most of the coconut specific operators since most of them contain at least one backslash (i.e. (?:<\*?)?(?!\.\.\.)\.\.(?:\*?>)?) )

Reply:
Replace the escape character with a double escape character, i.e. `\$` --> `\\$`.